elasticdl/python/data/parallel_transform.py

```diff
@@ -0,0 +1,99 @@
+import itertools
+from multiprocessing import Process, Queue
+
+
+def worker_loop(records, index_q, result_q, transform_fn):
+    """
+    The worker loop gets a index from the index queue, and
+    transform the records[index], then the result is put to
+    the result queue.
+
+    If a None index is got, the loop finishes.
+
+    :param records: a list object contains many records waiting to
+                    be transformed
+    :param index_q: a queue which generates the record index in records
+    :param result_q: a queue which stores the transform results
+    :param transform_fn: the transform function
+    """
+    while True:
+        index = index_q.get()
+        if index is None:
+            break
+        record = records[index]
+        res = transform_fn(record)
+        result_q.put((index, res))
+
+
+class ParallelTransform(object):
```

Comment:
Can we add an overall description for the ParellelTransform?

Reply:
Done